docs/docs/api-ref.md

```diff
@@ -5,257 +5,3010 @@ layout: docs
+`connection_credentials` | (Optional)  The credentials required to connect to the data source. The `ConnectionCredentials` object contains the authentication information for the data source (user name and password, and whether the credentials are embeded or OAuth is used). 
+ 
+
+
+**Exceptions**
+
+Error | Description  
+:--- | :--- 
+`File path does not lead to an existing file.`  |  Raises an error of the file path is incorrect or if the file is missing.
+`Invalid mode defined.`  |  Raises an error if the publish mode is not one of the defined options. 
+`Only .tds, tdsx, or .tde files can be published as datasources.`  |  Raises an error if the type of file specified is not supported.  
+
+
+**Returns**
 
-Deletes a workbook with the given ID.
+The `DatasourceItem` for the data source that was added or appened to. 
+
+
+**Example**
 
 ```py
-Workbooks.delete(id)
+
+  import tableauserverclient as TSC
+  server = TSC.Server('http://SERVERURL')
+    
+  ...
+
+  project_id = '3a8b6148-493c-11e6-a621-6f3499394a39'
+  file_path = 'C:\\temp\\WorldIndicators.tde'
+
+  # Use the project id to create new datsource_item
+  new_datasource = TSC.DatasourceItem(project_id)
+
+  # publish data source (specifed in file_path)
+  new_datasource = server.datasources.publish(
+                    new_datasource, file_path, 'CreateNew')
+
+    ...
 ```
 
-### Download Workbook
+<br>   
+<br>  
 
-Downloads a workbook to the specified directory.
+#### datasources.update
 
 ```py
-Workbooks.download(id, file_path)
+datasource.update(datasource_item)
 ```
 
-### Populate Views for a Workbook
+Updates the owner, or project of the specified data source. 
+
+REST API: [Update Datasource](http://onlinehelp.tableau.com/current/api/rest_api/en-us/help.htm#REST/rest_api_ref.htm#Update_Datasource%3FTocPath%3DAPI%2520Reference%7C_____79){:target="_blank"}
+
+**Parameters**
+
+Name   |  Description     
+ :--- | : ---    
+`datasource_item`  |  The `datasource_item` specifies the data source to update.
+
+
+
+**Exceptions**
+
+`Datasource item missing ID. Datasource must be retrieved from server first.` :  Raises an errror if the datasource_item is unspecified. Use the `Datasources.get()` method to retrieve that identifies for the data sources on the server.
+
+
+**Returns**
 
-Populates a list of views for a workbook object. You must populate views before you can iterate through the views.
+An updated `DatasourceItem`.
+
+
+**Example**
 
 ```py
-Workbooks.populate_views(workbook_obj)
+# from server-client-python/test/test_datasource.py
+
+    ...
+
+    single_datasource = TSC.DatasourceItem('test', '1d0304cd-3796-429f-b815-7258370b9b74')
+    # need to specify the ID - this is a workaround. 
```

Comment:
I don't like workaround in the docs :( why is this necessary? /cc @RussTheAerialist 

That is, what's this sample showing, do we need to demonstrate it that way